File: find_vi.js

const fs = require('fs');

const filename = "c:/Users/edama/Desktop/poly-lang/dataset_tokenized_modeB.json";

const content = fs.readFileSync(filename, 'utf-8');
const lines = content.split('\n');

for (let i = 0; i < lines.length; i++) {
    if (lines[i].includes('"lang": "vi"')) {
        console.log(`Found vi at line ${i + 1}`);
        console.log(lines.slice(i, i + 20).join('\n'));
        break;
    }
}

File: find_vi.py

import json

filename = "c:/Users/edama/Desktop/poly-lang/dataset_tokenized_modeB.json"

with open(filename, 'r', encoding='utf-8') as f:
    lines = f.readlines()

for i, line in enumerate(lines):
    if '"lang": "vi"' in line:
        print(f"Found vi at line {i+1}")
        # Print a few lines context
        print("".join(lines[i:i+20]))
        break
